hazelcast/src/main/java/com/hazelcast/config/MapConfig.java

```diff
@@ -796,6 +855,7 @@ public String toString() {
                 + ", mapAttributeConfigs=" + mapAttributeConfigs
                 + ", quorumName=" + quorumName
                 + ", queryCacheConfigs=" + queryCacheConfigs
+                + ", cacheDeserializedValues=" + cacheDeserializedValues
                 + '}';
```

Comment:
that's a good question. the `optimizeQuery` is not included in `hashCode` / `equals` calculation. cacheDeserializedValues is just an extension of the same idea so I didn't want to change hashCode / equals behaviour. @ahmetmircik: WDYT?